base/sparse/sparsematrix.jl

```diff
@@ -763,6 +763,11 @@ function sparse(B::Bidiagonal)
     return sparse([1:m;1:m-1],[1:m;2:m],[B.dv;B.ev], Int(m), Int(m)) # upper bidiagonal
 end
 
+function sparse(D::Diagonal{T}) where T
+    m = length(D.diag)
+    return SparseMatrixCSC(m, m, [1:m; m+1], [1:m;], convert(Vector{T}, D.diag))
```

Comment:
Perhaps `collect(1:(m+1))` in place of `[1:m; m+1]`, and likewise `collect(1:m)` in place of `[1:m;]`?
```julia
julia> m = 1000
1000

julia> @benchmark [1:$m; $m+1]
BenchmarkTools.Trial:
  memory estimate:  9.23 KiB
  allocs estimate:  46
  --------------
  minimum time:     5.816 μs (0.00% GC)
  median time:      6.441 μs (0.00% GC)
  mean time:        8.004 μs (9.65% GC)
  maximum time:     444.588 μs (93.81% GC)
  --------------
  samples:          10000
  evals/sample:     6

julia> @benchmark collect(1:$m)
BenchmarkTools.Trial:
  memory estimate:  7.97 KiB
  allocs estimate:  2
  --------------
  minimum time:     668.695 ns (0.00% GC)
  median time:      1.164 μs (0.00% GC)
  mean time:        2.173 μs (31.38% GC)
  maximum time:     24.659 μs (72.19% GC)
  --------------
  samples:          10000
  evals/sample:     131
```

Reply:
Interesting, will update.